### ISSUE TITLE
Can't install on python 3.13.0

### ISSUE THREAD

**Issue by spidychoipro** [yt-dlp/yt-dlp#11221]
### DO NOT REMOVE OR SKIP THE ISSUE TEMPLATE

- [X] I understand that I will be **blocked** if I *intentionally* remove or skip any mandatory\* field

### Checklist

- [X] I'm reporting a bug unrelated to a specific site
- [X] I've verified that I have **updated yt-dlp to nightly or master** ([update instructions](https://github.com/yt-dlp/yt-dlp#update-channels))
- [X] I've checked that all provided URLs are playable in a browser with the same IP and same login details
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/yt-dlp/yt-dlp/wiki/FAQ#video-url-contains-an-ampersand--and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched [known issues](https://github.com/yt-dlp/yt-dlp/issues/3766) and the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues **including closed ones**. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)

### Provide a description that is worded well enough to be understood

When I try to install it on python3.13.0 I can't

### Provide verbose output that clearly demonstrates the problem

- [X] Run **your** yt-dlp command with **-vU** flag added (`yt-dlp -vU <your command line>`)
- [ ] If using API, add `'verbose': True` to `YoutubeDL` params instead
- [X] Copy the WHOLE output (starting with `[debug] Command-line config`) and insert it below

### Complete Verbose Output

```shell
Collecting yt-dlp
  Using cached yt_dlp-2024.10.7-py3-none-any.whl.metadata (171 kB)
Collecting brotli (from yt-dlp)
  Using cached Brotli-1.1.0.tar.gz (7.4 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: certifi in c:\users\swpar\appdata\local\programs\python\python313\lib\site-packages (from yt-dlp) (2024.8.30)
Collecting mutagen (from yt-dlp)
  Using cached mutagen-1.47.0-py3-none-any.whl.metadata (1.7 kB)
Collecting pycryptodomex (from yt-dlp)
  Using cached pycryptodomex-3.21.0-cp36-abi3-win_amd64.whl.metadata (3.4 kB)
Requirement already satisfied: requests<3,>=2.32.2 in c:\users\swpar\appdata\local\programs\python\python313\lib\site-packages (from yt-dlp) (2.32.3)
Requirement already satisfied: urllib3<3,>=1.26.17 in c:\users\swpar\appdata\local\programs\python\python313\lib\site-packages (from yt-dlp) (2.2.3)
Collecting websockets>=13.0 (from yt-dlp)
  Using cached websockets-13.1-cp313-cp313-win_amd64.whl.metadata (7.0 kB)
Requirement already satisfied: charset-normalizer<4,>=2 in c:\users\swpar\appdata\local\programs\python\python313\lib\site-packages (from requests<3,>=2.32.2->yt-dlp) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in c:\users\swpar\appdata\local\programs\python\python313\lib\site-packages (from requests<3,>=2.32.2->yt-dlp) (3.10)
Using cached yt_dlp-2024.10.7-py3-none-any.whl (3.1 MB)
Using cached websockets-13.1-cp313-cp313-win_amd64.whl (159 kB)
Using cached mutagen-1.47.0-py3-none-any.whl (194 kB)
Using cached pycryptodomex-3.21.0-cp36-abi3-win_amd64.whl (1.8 MB)
Building wheels for collected packages: brotli
  Building wheel for brotli (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for brotli (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [11 lines of output]
      C:\Users\swpar\AppData\Local\Temp\pip-build-env-u8yvh_pr\overlay\Lib\site-packages\setuptools\_distutils\dist.py:261: UserWarning: Unknown distribution option: 'test_suite'
        warnings.warn(msg)
      running bdist_wheel
      running build
      running build_py
      creating bin\lib.win-amd64-cpython-313
      copying python\brotli.py -> bin\lib.win-amd64-cpython-313
      running build_ext
      <string>:70: DeprecationWarning: dep_util is Deprecated. Use functions from setuptools instead.
      building '_brotli' extension
      error: Microsoft Visual C++ 14.0 or greater is required. Get it with "Microsoft C++ Build Tools": https://visualstudio.microsoft.com/visual-cpp-build-tools/
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for brotli
Failed to build brotli
ERROR: ERROR: Failed to build installable wheels for some pyproject.toml based projects (brotli)
```

**Comment by seproDev** [comment]
The issue lies not with yt-dlp, but brotli (a dependency of yt-dlp) not yet providing prebuilt Python 3.13 wheels.
Either wait for brotli to provide prebuilt wheels for python 3.13, install the build tools as the error message tells you to and build the wheels yourself, or install yt-dlp without brotli.

**Comment by gamer191** [comment]
(this is a duplicate of https://github.com/google/brotli/issues/1208, by the way)

**Comment by PalmtopTiger** [comment]
yt-dlp can't be installed without brotli, because this package is a mandatory dependency (for cpython). Moreover, yt-dlp can work with brotlicffi.
Is it possible to make brotli an optional dependency? Google may not build wheels for a long time. And each time it takes longer.

**Comment by seproDev** [comment]
brotli is an optional dependency of yt-dlp. In fact all dependencies are optional. The functionality of yt-dlp may be limited, but it will still run.
Currently brotli gets installed with `pip install yt-dlp` due to being a recommended dependency. In due time we will remove brotli and only install it (alongside other depnendecies) when `pip install yt-dlp[default]` is used.
Even without that you can install yt-dlp through pip without brotli by using `--no-deps`.

**Comment by PalmtopTiger** [comment]
@seproDev thanks for the reply.
When running `pip install yt-dlp[default]`, as well as just `pip install yt-dlp`, pip tries to find a version that doesn't depend on brotli, and ends up installing yt-dlp-2022.2.4.
`pip install yt-dlp --no-deps` works. But when upgrading all packages, it still installs the old version.

**Comment by bashonly** [comment]
Currently, `[default]` is an empty group: we added it months ago so that we could update our documentation in advance to instruct/condition users to install yt-dlp with the `default` group/extra.

When #11255 is merged, all dependencies will be truly optional and they will only be installed if the `[default]` group is added to `yt-dlp` in the pip/pipx install command.

**Comment by grqz** [comment]
https://github.com/google/brotli/issues/1208#issuecomment-2422460041

They've built and uploaded the wheels for Python 3.13. Should this issue be closed? Or is it better to merge #11255 before this is closed?
